test/extended/images/mysql_replica.go

```diff
@@ -0,0 +1,266 @@
+package images
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+	"time"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/labels"
+	"k8s.io/kubernetes/pkg/util/wait"
+)
+
+func selector(selector string) labels.Selector {
+	ret, err := labels.Parse(selector)
+	if err != nil {
+		g.Fail(fmt.Sprintf("Couldn't parse selector %v: %v", selector, err))
+	}
+	return ret
+}
+
+// getPods looks up pods with "oc get" applying a label selector and returns
+// names of matching pods
+func getPods(oc *exutil.CLI, label labels.Selector) (pods []string, err error) {
+	podList, err := oc.KubeREST().Pods(oc.Namespace()).List(label, nil)
+	if err != nil {
+		return nil, err
+	}
+	for _, pod := range podList.Items {
+		if pod.Status.Phase == kapi.PodRunning {
+			pods = append(pods, pod.Name)
+		}
+	}
+	return pods, nil
+}
+
+// pollForPods will call getPods in a loop until the required number of pods
+// is returned
+func pollForPods(oc *exutil.CLI, label labels.Selector, count int, timeout time.Duration) ([]string, error) {
+	var pods []string
+	err := wait.Poll(1*time.Second, timeout, func() (bool, error) {
+		p, e := getPods(oc, label)
+		if e != nil {
+			return true, e
+		}
+		if len(p) < count {
+			return false, nil
+		}
+		pods = p
+		return true, nil
+	})
+	return pods, err
+}
+
+type mysql struct {
+	oc      *exutil.CLI
```

Comment:
I'm OK with creating helper structs like this, but don't like bundling the exutil.CLI here. Is there a way to make this more generic? And don't bundle *exutil.CLI ? You really just need PodInterface here and not the full exutil.CLI